### ISSUE TITLE
Pool value representation does not add up

### ISSUE THREAD

**Issue by kremalicious** [oceanprotocol/market#1208]
Let's look at this pool which is being looked at with the account who created the pool:

`did:op:d855538c659038bedbdb094417391ac63a9d22084900cf510bb78ed66f2ac6da`
https://rinkeby.etherscan.io/address/0x7807e6ac970fd11cac8e491b711b70c5187aa9ee
<img width="496" alt="Screen Shot 2022-03-17 at 14 09 34" src="https://user-images.githubusercontent.com/90316/158825838-fad90cf6-87c5-4f8e-84da-22b53d688992.png">

It was created by adding 50 OCEAN from user wallet:
https://rinkeby.etherscan.io/tx/0x9465e0f040323faedfa94ea0e938cab382c9a9ea60c7ba831ef2e868c6c3ee5e

But numbers suggest the total pool TVL doubled. No further transactions happened on that pool.

If I get this right, user value locked & owner value locked is correct, but total TVL should be simply half, so also 50.

**Comment by mihaisc** [comment]
We may be doubling things too much 😁

**Comment by EnzoVezzaro** [comment]
yeah, this is caused by [this line](https://github.com/oceanprotocol/market/blob/fix/issue-1208-wrong-pool-tvl-value/src/@context/Pool/index.tsx#L123). I'm still trying to figure out why is this calculation being make, what is used for?

**Comment by kremalicious** [comment]
this calculation seems overcomplicated and a remnant of v3 where datatoken has value based on pool spot price. But if I'm not mistaken all we need to do now is 

```js
const totalLiquidityInOcean = new Decimal(poolData.baseTokenLiquidity).mul(2)
```

as we have 50/50 weight pools and "staking bot" always keeps value of base token & datatoken in sync

**Comment by EnzoVezzaro** [comment]
> this calculation seems overcomplicated and a remnant of v3 where datatoken has value based on pool spot price. But if I'm not mistaken all we need to do now is
> 
> ```js
> const totalLiquidityInOcean = new Decimal(poolData.baseTokenLiquidity).mul(2)
> ```
> 
> as we have 50/50 weight pools and "staking bot" always keeps value of base token & datatoken in sync

Mmm.... tried, but I get the same issue. The only way it works is by leaving `new Decimal(poolData.baseTokenLiquidity)` by itselft.

`const totalLiquidityInOcean = isValidNumber(poolData.spotPrice)
      ? new Decimal(poolData.baseTokenLiquidity)
      : new Decimal(0)`

this way I get these results:

did:op:d9efba2812840e7de6ca6e349f63942da3e3cbbf8ad143e2bf84d768ffc904e7
<img width="347" alt="Screen Shot 2022-03-24 at 8 54 42 AM" src="https://user-images.githubusercontent.com/13741335/159920952-074baba2-a6e4-4ada-b76e-23f8db06a0ef.png">

did:op:f66b072ec293aa76a29668df9ba7dbaf2cb0bb610f3475f8ebeebcfb731749b8
<img width="350" alt="Screen Shot 2022-03-24 at 8 54 13 AM" src="https://user-images.githubusercontent.com/13741335/159920959-3fce5e44-a5f0-489b-860d-50f89cb5f743.png">

did:op:c2a31770d4e6a8fde05a8a90e30fa98b35443b21f7ef155baedd2cd0a5fb3fe8
<img width="358" alt="Screen Shot 2022-03-24 at 8 53 56 AM" src="https://user-images.githubusercontent.com/13741335/159920961-39dbda45-6f8a-4995-a193-2930e99d68ad.png">

But I'm afraid of missing something because this one is not correct:
did:op:2c33b5fea530508a9f019a1dab89122a10b51503b7947db02513cfc6dd511f65
<img width="362" alt="Screen Shot 2022-03-24 at 8 58 28 AM" src="https://user-images.githubusercontent.com/13741335/159921398-874fab14-7434-4951-aef3-435153a4f63a.png">

I'm still investigating why

**Comment by EnzoVezzaro** [comment]
Maybe I'm misunderstanding the UI here, but in this example: 

did:op:2c33b5fea530508a9f019a1dab89122a10b51503b7947db02513cfc6dd511f65
<img width="358" alt="Screen Shot 2022-03-24 at 9 48 49 AM" src="https://user-images.githubusercontent.com/13741335/159930585-69480e2e-2014-4a3c-a853-33a923a799f6.png">

this user hasn't provided liquidity to the pool, but it Swapped 1.065 Ocean. 

The values shown doesn't seem to reflect this, and there are some values that doesn't adds up (at least for me, I'm not sure). For example:

the `Owner Value Locked` is `89.78%` of the pool. So, I'm guessing the remaining `10.22%` is the Swap done by this user + gas fees (maybe). This doesn't adds up as: 

TVL: 60.777
Owner Value Locked: 54.5
Remaining: 6.277

The user swapped 1.065 Ocean

So, after removing the tokens swapped, the remaining is: 5.212 Ocean Tokens are basically lost or consumed somewhere.

Maybe I'm missing something, can you please clarify this for me

**Comment by mihaisc** [comment]
Well there are swap/opc fees so take those in consideration.

**Comment by EnzoVezzaro** [comment]
Here is my test, I'm sorry if it's quite long 😅 I tried to be as precise as I could to be able to test well this use case.

Test Actions on Pool

User 1:
Created Pool (OVL): 500 OT (ocean tokens)

Showed (OVL): 500 OT (ocean tokens)
Added: 10 OT
Remove: 1.044
Added: 20 OT

All transactions showed the correct amounts (of course counting the fees which seems alright)

Expected (Your Value Locked): 531.044
Showed (Your Value Locked): 519.657
Diff: 11.387 // fair enough

Showed TVL: 519.657

<img width="327" alt="Screen Shot 2022-03-29 at 6 32 47 AM" src="https://user-images.githubusercontent.com/13741335/160608717-134222e8-0635-4d91-9505-f25344f48184.png">

______

User 2: 
Initial (TVL): 519.657
Initial (YVL - your value locked): 0
Initial (OVL - owner value locked): 519.657
Added: 100 OT

Expected (Your Value Locked): 100
Showed (Your Value Locked): 96.247
Diff: 3.753 OT // fair enough

Expected TVL: 619.657
Showed TVL: 619.657

<img width="333" alt="Screen Shot 2022-03-29 at 6 43 55 AM" src="https://user-images.githubusercontent.com/13741335/160608819-d3f3f2fc-de5f-4318-86fc-938c016e9862.png">

Here is weird, when adding the TVL align perfectly: I deposit 100, I got 100 (529.657 + 100 = 619.657 no fees diff). So, it's unclear what happened with the 3.753 OT, I'd expect them to be deducted from the TVL.

________

User 2:
Initial (TVL): 619.657
Initial (YVL): 94.247
Initial (OVL): 523.410
Remove: 50.939

Expected (Your Value Locked): 43.308
Showed (Your Value Locked): 41.477
Diff: 1.831 // fair enough

Expected TVL: 568.718
Showed TVL: 566.037
Diff: 2.681 // fair enough, but why here we have a diff (probably fees) while by adding liquidity we don't?

<img width="330" alt="Screen Shot 2022-03-29 at 7 27 40 AM" src="https://user-images.githubusercontent.com/13741335/160609067-03621146-9a60-40fa-b2fd-eb275390252e.png">

Percentages align perfectly so far.
________

User 2: 
Initial (TVL): 566.037
Initial (YVL): 41.477
Initial (OVL): 524.560
Trade: 100 OT to 8.31884 BRAPOR-8
Transaction: https://rinkeby.etherscan.io//tx/0x6bf782690a645ec002f48b864d61494aa4461cec36405ab769c0c8be7868d959

Initial (Your Value Locked): 41.477
Showed (Your Value Locked): 48.798
Diff: +7.321 // gains - which sounds about right (7.32% of 100 OT)

Initial (OVL): 524.560
Showed (0VL): 617.140
Diff: +92.580 // gains - which sounds about right (92.67% of 100 OT)

Expected TVL: 668.718
Showed TVL: 665.937
Diff: 2.781 // fair enough

_______

Here is where is a bit tricky.

User 3:
Initial (TVL): 655.937
Initial (YVL): 0
Initial (OVL): 617.140
Trade: 200 OT to 10.88059 BRAPOR-8

Initial (Your Value Locked): 0
Showed (Your Value Locked): 0
Diff: 0

Initial (OVL): 617.140
Showed (0VL): 802.299
Diff: +185.159 // gains - which sounds about right (92.67% of 200 OT)

Expected TVL: 865.937
Showed TVL: 865.737
Diff: 0.2 // weirdly low, but ok

<img width="339" alt="Screen Shot 2022-03-29 at 8 07 37 AM" src="https://user-images.githubusercontent.com/13741335/160609687-5699784b-fb71-4bd4-a5c1-1d860b7974da.png">

In this case the user's actually sees the difference in price, and also sees his transaction in the history. But in the Pool tab is not indicated anywhere that the user is contributing to this pool. Most probably is correct, but I'd expect to see somewhere that I have contributed to this pool by trading.

Besides this, the rest seems fine to me. All numbers seem aligned, I don't know if it's worth it testing more this to make sure all cases are covered.
